### PR TITLE
fix: Infer correct number of components in gaze init

### PR DESCRIPTION
**Pull request by dkrako**
## Description

This PR implements a more robust way of inferring the number of components in a GazeDataFrame.

Fixes issue #514

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- [x] Added components checks to existing gaze init tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/pymovements/gaze/gaze_dataframe.py [hunk 1]
```diff
@@ -526,3 +504,48 @@ def _check_n_components(n_components: Any) -> None:
...
+    -------
+    int or None
+        Number of components
+
+    Raises
+    ------
+    ValueError
+        If number of components is not equal for all considered columns and rows.
+    """
+    all_considered_columns = ['pixel', 'position', 'velocity', 'acceleration']
```
Comment (prassepaul) [inline src/pymovements/gaze/gaze_dataframe.py:533]: maybe we can put all_considered_columns in the arg list? maybe we extend the list of columns in the future, or someone wants to use additional (project) specific columns?
We can set the default to the predefined list of columns.

**Review by prassepaul — CHANGES_REQUESTED** [review]
see the comment above. maybe we can/should change that?

File: src/pymovements/gaze/gaze_dataframe.py (hunk 1 above)
Comment (dkrako) [reply src/pymovements/gaze/gaze_dataframe.py:533]: Adding some kind of support for non-standard columns in the future is a good idea, but that's actually a feature request.
As that would be out of scope of this PR which is a bugfix, I have created a new issue for this were we can discuss details: #524

Comment (prassepaul) [reply src/pymovements/gaze/gaze_dataframe.py:533]: cool. thx

Comment (dkrako) [reply src/pymovements/gaze/gaze_dataframe.py:533]: Also, keep in mind that setting lists (or any other mutable datatype) as default arguments in functions introduces horrible bugs in python:

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

**Review by prassepaul — APPROVED** [review]

**Review by prassepaul — APPROVED** [review]

**Pull request merged**